server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/sql/Db2SqlVisitor.java

```diff
@@ -106,7 +118,11 @@ private List<Node> createHostVariableDefinitionNode(ParserRuleContext ctx, Parse
                         getLocality(this.context.getExtendedDocument().mapLocation(constructRange(nameCtx)))))
                 .usageClauses(ImmutableList.of(UsageFormat.DISPLAY))
                 .build();
-
+        if (ctx instanceof  Db2SqlParser.Lob_host_variablesContext
+                && ((Db2SqlParser.Lob_host_variablesContext) ctx).lobWithSize() != null
+                && ((Db2SqlParser.Lob_host_variablesContext) ctx).lobWithSize().BINARY() != null) {
+            generateVarbinVariables(variableDefinitionNode, ((Db2SqlParser.Lob_host_variablesContext) ctx).lobWithSize().dbs_integer().getText(), ctx);
```

Comment:
Just a thought, this puts extra responsibility to `createHostVariableDefinitionNode` . Should this go to the actual visit method that calls this method ?